src/Microsoft.IdentityModel.Xml/DsigSerializer.cs

```diff
@@ -140,6 +145,9 @@ public virtual KeyInfo ReadKeyInfo(XmlReader reader)
                             }
                         }
 
+                        var X509Data = new X509Data(issuerSerial, subjectName, SKI, certificateData, CRL);
+                        keyInfo.X509Data.Add(X509Data);
+
                         // </X509Data>
                         reader.ReadEndElement();
```

Comment:
We need to check if X509Data is an empty element. If so we should throw as that indicates an error.

Reply:
Done!